message/message.go

```diff
@@ -301,6 +321,27 @@ func (gsr GraphSyncResponse) ExtensionNames() []graphsync.ExtensionName {
 	return extNames
 }
 
+func (gsr GraphSyncResponse) Metadata() graphsync.LinkMetadata {
+	return GraphSyncLinkMetadata{gsr.metadata}
+}
+
+func (gslm GraphSyncLinkMetadata) Iterate(iter graphsync.LinkMetadataIterator) {
+	for _, md := range gslm.linkMetadata {
+		iter(md.Link, md.Action)
+	}
+}
+
+func (gslm GraphSyncLinkMetadata) Clone() []GraphSyncLinkMetadatum {
```

Comment:
I wonder if this should skip the copy -- it should only be used on the internal type -- it's not exposed for the interface, and the only place I see it used is simply to go into the serialized message where it's not otherwise modified. (and also in testing)